### PR TITLE
Scheduling

### PR DESCRIPTION
**Pull request by Pentadrago**
The harvester team moved the persistence in a module called packrat and did some cleanup.

### CODE REVIEW

**Comment by RudiKlassen** [comment]
@matthiasbalke Please merge the pullrequest.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Mon Mar 14 18:26:05 CET 2016
+#Wed Apr 13 20:17:35 CEST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-all.zip
```
Comment (matthiasbalke) [inline gradle/wrapper/gradle-wrapper.properties]: Is there a benefit in switching to the larger gradlew `all` distribution?

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/config/UpdatePluginConfigJob.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * reads the configurations of the sources and schedules the corresponding jobs
+ */
+public class UpdatePluginConfigJob implements Job {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdatePluginConfigJob.class);
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("executing update");
+        // SourceConfig config = new SourceConfig("widget", "urlsource", new Date(), 5, new HashMap<>());
```
Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/config/UpdatePluginConfigJob.java]: commented out code, is bad practice. please remove or finish the code before merging.

File: harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/config/UpdatePluginConfigJob.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        JobDataMap configuration = context.getJobDetail().getJobDataMap();
+        ApplicationContext applicationContext = (ApplicationContext) configuration.get(SchedulingService.PARAM_CONTEXT);
+        SourceConfigRepository repository = applicationContext.getBean(SourceConfigRepository.class);
+        // repository.save(config);
```
Comment (matthiasbalke) [inline harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/source/config/UpdatePluginConfigJob.java]: commented out code, is bad practice. please remove or finish the code before merging.

File: harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/source/url/UrlSourceRetrieveJobTest.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.github.reflectoring.infiniboard.harvester.source.url;
+
+import org.junit.Test;
+
+public class UrlSourceRetrieveJobTest {
+
+    @Test
+    public void saveToDB() {
```
Comment (matthiasbalke) [inline harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/source/url/UrlSourceRetrieveJobTest.java]: This test seems to not do anything. Please remove or complete the test before merging.

File: packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java [hunk 5]
```diff
@@ -7,17 +7,22 @@
...
 
     @Id
     private String url;
+
     private Date lastFetched;
+
     private String content;
 
-    public UrlData(String url, Date lastFetched, String content) {
+    private Integer statusCode;
```
Comment (matthiasbalke) [inline packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java]: let's use primitive data types, as long as we can avoid having null values. Is it possible here to use `int`?

File: packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java [hunk 6]
```diff
@@ -7,17 +7,22 @@
...
+
+    public UrlSource(String url, Date lastFetched, String content, Integer statusCode) {
```
Comment (matthiasbalke) [inline packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java]: see above: `Integer` -> `int`

File: packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java [hunk 7]
```diff
@@ -31,4 +36,8 @@ public Date getLastFetched() {
     public String getContent() {
         return content;
     }
+
+    public Integer getStatusCode() {
```
Comment (matthiasbalke) [inline packrat/src/main/java/com/github/reflectoring/infiniboard/packrat/source/UrlSource.java]: see above: `Integer` -> `int`

File: packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/OverseerTestApplication.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * Spring Boot application for overseer tests
+ */
+@SpringBootApplication
+public class OverseerTestApplication {
+
+}
```
Comment (matthiasbalke) [inline packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/OverseerTestApplication.java]: Let's always end files with OR without newline and document this style in our wiki. I prefer ending with a newline. But let us discuss this next meeting.

File: packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        attributes.put("alpha", "one");
+        attributes.put("beta", "two");
+        attributes.put("gamma", "three");
+        SourceConfig config = new SourceConfig("foo", "bar", new Date(), 5 * 60, attributes);
+
+        repository.save(config);
+
+        List<SourceConfig> configList = repository.findBySourceId("bar");
+        assertThat(configList).hasSize(1);
+        SourceConfig sourceConfig = configList.get(0);
```
Comment (matthiasbalke) [inline packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java]: this test asserts two conditions. please refactor it into two tests each asserting one condition.

File: packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        assertThat(sourceConfig.getConfigData()).containsKeys("alpha", "beta", "gamma");
+    }
+
+}
```
Comment (matthiasbalke) [inline packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java]: see above: no newline at end

File: test-base/build.gradle [hunk 11]
```diff
@@ -1,5 +1,5 @@
 dependencies {
     compile(
-            'org.assertj:assertj-core:3.3.0'
+            "org.assertj:assertj-core:${version_assertj}",
     )
 }
```
Comment (matthiasbalke) [inline test-base/build.gradle]: see above: no newline at end

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Pentadrago) [inline gradle/wrapper/gradle-wrapper.properties]: this was recommeded by IDEA for better Gradle integration

File: packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java (hunk 9 above)
Comment (RudiKlassen) [inline packrat/src/test/java/com/github/reflectoring/infiniboard/packrat/source/SourceConfigTest.java]: Or remove the first assertion. The Test will fail, if the list is empty.

**Pull request merged**